### PR TITLE
Users/arjgupta/iis mg update2

### PR DESCRIPTION
**Pull request by arjgupta**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts [hunk 1]
```diff
@@ -7,9 +7,10 @@ var msDeployUtility = require('webdeployment-common/msdeployutility.js');
 var zipUtility = require('webdeployment-common/ziputility.js');
 var utility = require('webdeployment-common/utility.js');
 var msDeploy = require('webdeployment-common/deployusingmsdeploy.js');
-var jsonSubstitutionUtility = require('webdeployment-common/jsonvariablesubstitutionutility.js');
-var xmlSubstitutionUtility = require('webdeployment-common/xmlvariablesubstitutionutility.js');
-var xdtTransformationUtility = require('webdeployment-common/xdttransformationutility.js');
+var ft = require('webdeployment-common/fileTransformationsUtility.js');
+//var jsonSubstitutionUtility = require('webdeployment-common/jsonvariablesubstitutionutility.js');
```
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts]: can u remove the comments ?

File: Tasks/Common/webdeployment-common/fileTransformationsUtility.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        if(tl.osType().match(/^Win/)) {
+            var transformConfigs = ["Release.config"];
+            if(environmentName) {
+                transformConfigs.push(environmentName + ".config");
+            }
+            xdtTransformationUtility.basicXdtTransformation(folderPath, transformConfigs);  
+            console.log(tl.loc("XDTTransformationsappliedsuccessfully"));
+        }
+        else {
+            throw new Error(tl.loc("CannotPerformXdtTransformationOnNonWindowsPlatform"));
```
Comment (vincentdass) [inline Tasks/Common/webdeployment-common/fileTransformationsUtility.ts]: error messages are missing in task.json

File: Tasks/Common/webdeployment-common/fileTransformationsUtility.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        }
+    }
+
+    if(xmlVariableSubstitution) {
+        await xmlSubstitutionUtility.substituteAppSettingsVariables(folderPath);
+        console.log(tl.loc('XMLvariablesubstitutionappliedsuccessfully'));
+    }
+
+    if(JSONFiles.length != 0) {
+        jsonSubstitutionUtility.jsonVariableSubstitution(folderPath, JSONFiles);
```
Comment (vincentdass) [inline Tasks/Common/webdeployment-common/fileTransformationsUtility.ts]: error messages are missing in task.json - > (inside jsonVariableSubstitution )

File: Tasks/IISWebAppDeploymentOnMachineGroup/make.json [hunk 4]
```diff
@@ -13,6 +13,11 @@
                 "archiveName": "MSDeploy.zip",
                 "url": "https://vstsagenttools.blob.core.windows.net/tools/MSDeploy/3.6/MSDeploy.zip",
                 "dest": "./"
+            },
+			{
```
Comment (vincentdass) [inline Tasks/IISWebAppDeploymentOnMachineGroup/make.json]: indentation.

File: Tasks/IISWebAppDeploymentOnMachineGroup/deployiiswebapp.ts [hunk 5]
```diff
@@ -18,7 +19,11 @@ async function run()
 		var excludeFilesFromAppDataFlag: boolean = tl.getBoolInput('ExcludeFilesFromAppDataFlag', false);
 		var takeAppOfflineFlag: boolean = tl.getBoolInput('TakeAppOfflineFlag', false);
 		var additionalArguments: string = tl.getInput('AdditionalArguments', false);
+        var xmlTransformation: boolean = tl.getBoolInput('XmlTransformation', false);
```
Comment (vincentdass) [inline Tasks/IISWebAppDeploymentOnMachineGroup/deployiiswebapp.ts]: indentation.

File: Tasks/IISWebAppDeploymentOnMachineGroup/deployiiswebapp.ts [hunk 6]
```diff
@@ -32,6 +37,13 @@ async function run()
 		webDeployPkg = availableWebPackages[0];
 
 		var isFolderBasedDeployment = await utility.isInputPkgIsFolder(webDeployPkg);
+
+        if(JSONFiles.length != 0 || xmlTransformation || xmlVariableSubstitution) {
```
Comment (vincentdass) [inline Tasks/IISWebAppDeploymentOnMachineGroup/deployiiswebapp.ts]: indentation.

File: Tasks/IISWebAppDeploymentOnMachineGroup/deployiiswebapp.ts [hunk 7]
```diff
@@ -4,6 +4,7 @@ import fs = require('fs');
 
 var msDeploy = require('webdeployment-common/deployusingmsdeploy.js');
 var utility = require('webdeployment-common/utility.js');
+var ft = require('webdeployment-common/fileTransformationsUtility.js');
```
Comment (vincentdass) [inline Tasks/IISWebAppDeploymentOnMachineGroup/deployiiswebapp.ts]: can u name the variable like 'fileTransformationsUtility' ?

File: Tasks/IISWebAppDeploymentOnMachineGroup/task.json [hunk 8]
```diff
@@ -23,6 +23,11 @@
   "minimumAgentVersion": "1.102.0",
```
Comment (vincentdass) [inline Tasks/IISWebAppDeploymentOnMachineGroup/task.json]: Agent version should be changed to 2.104.1.

File: Tasks/Common/webdeployment-common/fileTransformationsUtility.ts (hunk 2 above)
Comment (arjgupta) [inline Tasks/Common/webdeployment-common/fileTransformationsUtility.ts]: have updated task.json accordingly

File: Tasks/Common/webdeployment-common/fileTransformationsUtility.ts (hunk 3 above)
Comment (arjgupta) [inline Tasks/Common/webdeployment-common/fileTransformationsUtility.ts]: have updated task.json accordingly

**Comment by mmulhearn** [comment]
There is an issue on Ln 17.  Visual Studio creates transforms as Web.<Configuration Name>.config.  Line 17 is adding the environment config to the transformation array as <Environment Name>.config, which is not the same as Web.<Environment Name>.config.  The 3.0 release task uses this library and my environmental configs are not being fired and I believe it's because of this.

**Pull request merged**